arangod/IResearch/IResearchFilterFactory.cpp

```diff
@@ -3570,17 +3569,68 @@ Result fromFuncStartsWith(
 
     TRI_ASSERT(filterCtx.analyzer);
     kludge::mangleField(name, filterCtx.analyzer);
-    filter->boost(filterCtx.boost);
+
+    // Try to optimize us away
+    if (!isMultiPrefix && !prefixes.empty() &&
+         ctx.allowFiltersMerge && filter->type() == irs::type<irs::And>::id()) {
+      for (auto& f : *filter) {
```

Comment:
I'd opt for moving this optmization logic into a separate cpp unit so we can start collecting such optimization for further backporting to iresaearch